mitodl/open-discussions
# Add support anonymous users in search, and add support for public channels

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #1483 

#### What's this PR do?
 - Adds support for anonymous users in search
 - Adds support for public and restricted channels that a user is not already a moderator of contributor for

#### How should this be manually tested?
Find a restricted or public channel. Remove your user as a contributor for that channel, then view the channel. It should work fine for `master` and this PR.

View the channel search page. On `master` it should show no results but you should see results in this PR.

Open an incognito window and view the search. You should see the same results as you do when you're logged in.
